GVFS/GVFS.Common/GVFSPlatform.cs

```diff
@@ -87,20 +87,28 @@ public bool TryGetNormalizedPathRoot(string path, out string pathRoot, out strin
             return true;
         }
 
-        public class GVFSPlatformConstants
+        public abstract class GVFSPlatformConstants
         {
             public static readonly char PathSeparator = Path.DirectorySeparatorChar;
+            public abstract string ExecutableExtension { get; }
+            public abstract string InstallerExtension { get; }
+
+            public virtual string GVFSBinDirectoryPath
+            {
+                get
+                {
+                    return Path.Combine(

```

Comment:
Is this implementation Windows specific?  

I'm wondering if this class should be changed to be an interface instead of an class, will any of the implementations shared?

Reply:
Sorry I missed that we still have a bunch of non-virtual helper methods (e.g. `GVFSHooksExecutableName`).  In that case, perhaps just make  `GVFSBinDirectoryPath`, `GVFSBinDirectoryName`, and `GVFSExecutableName` abstract as they appear to be implementation specific?

Reply:
makes sense. Updated.